index.js

```diff
@@ -32,6 +32,11 @@ export class Item {
       return false
     }
 
+    // Prevent broken pointers
+    if (this === item) {
+      return false
+    }
```

Comment:
Oh btw, better: move the two ifs-that-return-false into each other!

```
// If self is detached or prepending ourselves, return false.
if (!list || this === item) {
  return false
}
```

What do you think? Sorry I didn’t see this before!

Reply:
Works for me!